vm/vm.go

```diff
@@ -0,0 +1,121 @@
+package vm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/kubearmor/kubearmor-client/k8s"
+	pb "github.com/kubearmor/kubearmor-client/protobuf"
+	"github.com/rs/zerolog/log"
+	"google.golang.org/grpc"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type VmOptions struct {
+	IP     string
+	VMName string
+	File   string
+}
+
+var (
+	serviceAccountName = "kvmsoperator"
+	pbClient           pb.HandleCliClient
+)
+
+func initGrpcClient(ip string) error {
+	// Connect to gRPC server
+	grpcClientConn, err := grpc.DialContext(context.Background(), net.JoinHostPort(ip, "32770"), grpc.WithInsecure())
+	if err != nil {
+		return err
+	}
+
+	pbClient = pb.NewHandleCliClient(grpcClientConn)
+	if pbClient == nil {
+		return errors.New("invalid grpc client handle")
+	}
+	return nil
+}
+
+func writeScriptDataToFile(options VmOptions, scriptData string) error {
+
+	var filename string
+
+	if options.File == "none" {
+		filename = options.VMName + ".sh"
+	} else {
+		filename = options.File
+	}
+
+	file, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	_, err = file.WriteString(scriptData)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("VM installation script copied to %s\n", filename)
+	return nil
+}
+
+func getClusterIP(c *k8s.Client, options VmOptions) (string, error) {
+
+	var clusterIP string
+
+	svcList, err := c.K8sClientset.CoreV1().Services("kube-system").List(context.Background(), metav1.ListOptions{
+		FieldSelector: "metadata.name=" + serviceAccountName})
+	if err != nil {
+		return "", err
+	}
+
+	for _, svc := range svcList.Items {
+		clusterIP = svc.Spec.ClusterIP
+		break
+	}
+
+	if options.IP != "none" {
+		return options.IP, err
+	}
+
+	return clusterIP, err
+}
+
+func FileDownload(c *k8s.Client, options VmOptions) error {
+
+	// Check if kvmsoperator is up and running
+	if _, err := c.K8sClientset.CoreV1().ServiceAccounts("kube-system").Get(context.Background(), serviceAccountName, metav1.GetOptions{}); err != nil {
```

Comment:
The kube-system namespace should not be hardcoded. It is possible for kubearmor to be installed as part of some other namespace.

Reply:
I will check for other possible solution.